File: 6909-longest-even-odd-subarray-with-threshold/6909-longest-even-odd-subarray-with-threshold.py
class Solution:
    def longestAlternatingSubarray(self, nums: List[int], threshold: int) -> int:
        l, r = 0, 0
        res = 0
        
        while(r < len(nums)):
            x = l - r
            
            if x % 2 == nums[r] % 2 and nums[r] <= threshold:
                res = max(res, r-l+1)
            else:
                if nums[r] % 2 == 0 and nums[r] <= threshold:
                    l = r
                else:
                    l = r + 1
            r += 1 
        
        return res
        
        
      
        
    

File: 6909-longest-even-odd-subarray-with-threshold/6909-longest-even-odd-subarray-with-threshold.rs
impl Solution {
    pub fn longest_alternating_subarray(nums: Vec<i32>, threshold: i32) -> i32 {
        let n = nums.len();
        let mut res = 0;
        
        for i in 0..n {
            for j in i..n {
                if nums[j] > threshold {
                    break;
                }
                
                if (j - i) % 2 == 1 && nums[j] % 2 == 0 {
                    break;
                }
                
                if (j - i) % 2 == 0 && nums[j] % 2 == 1 {
                    break;
                }
                
                res = res.max(j - i + 1);
            }
        }
        
        res as i32
    }
}